drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -0,0 +1,439 @@
+			pcr1 |= MCHP_GPIO_CTRL_IDET_BEDGE;
+		}
+		sys_write32(pcr1, pcr1_addr);
+		mask = 0u;
+		/* HW takes several clocks to stabilize the first time
+		 * edge detect is enabled.
+		 */
+		for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
+			mask |= sys_read32(pcr1_addr);
+		}
+		/* trick the compiler by using mask. write to read-only */
+		sys_write32(mask, XEC_GPIO_DLY_ADDR);
+	} else {
+		pcr1 |= MCHP_GPIO_CTRL_IDET_DISABLE;
+		sys_write32(pcr1, pcr1_addr);
+	}
+
+	mchp_soc_ecia_girq_src_clr(config->girq_id, pin);
+
+	gpio_xec_intr_en(pin, mode, config->girq_id);
+
+	return 0;
+}
+
+static int gpio_xec_port_set_masked_raw(const struct device *dev,
+					uint32_t mask,
+					uint32_t value)
+{
+	uintptr_t pout_addr = pin_parout_addr(dev);
+
+	xec_mask_write32(pout_addr, mask, value);
+
+	return 0;
+}
+
+static int gpio_xec_port_set_bits_raw(const struct device *dev, uint32_t mask)
+{
+	uintptr_t pout_addr = pin_parout_addr(dev);
+
+	sys_write32(sys_read32(pout_addr) | mask, pout_addr);
+
+	return 0;
+}
+
+static int gpio_xec_port_clear_bits_raw(const struct device *dev,
+					uint32_t mask)
+{
+	uintptr_t pout_addr = pin_parout_addr(dev);
+
+	sys_write32(sys_read32(pout_addr) & ~mask, pout_addr);
+
+	return 0;
+}
+
+static int gpio_xec_port_toggle_bits(const struct device *dev, uint32_t mask)
+{
+	uintptr_t pout_addr = pin_parout_addr(dev);
+
+	sys_write32(sys_read32(pout_addr) ^ mask, pout_addr);
+
+	return 0;
+}
+
+static int gpio_xec_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	uintptr_t pin_addr = pin_parin_addr(dev);
+
+	*value = sys_read32(pin_addr);
+
+	return 0;
+}
+
+static int gpio_xec_manage_callback(const struct device *dev,
+				    struct gpio_callback *callback, bool set)
+{
+	struct gpio_xec_data *data = dev->data;
+
+	gpio_manage_callback(&data->callbacks, callback, set);
+
+	return 0;
+}
+
+static void gpio_gpio_xec_port_isr(const struct device *dev)
+{
+	const struct gpio_xec_config *config = dev->config;
+	struct gpio_xec_data *data = dev->data;
+	uint32_t girq_result;
+
+	/* Figure out which interrupts have been triggered from the EC
+	 * aggregator result register
+	 */
+	girq_result = mchp_soc_ecia_girq_result(config->girq_id);
+
+	/* Clear source register in aggregator before firing callbacks */
+	mchp_soc_ecia_girq_src_clr_bitmap(config->girq_id, girq_result);
+
+	gpio_fire_callbacks(&data->callbacks, dev, girq_result);
+}
+
+static const struct gpio_driver_api gpio_xec_driver_api = {
+	.pin_configure = gpio_xec_configure,
+	.port_get_raw = gpio_xec_port_get_raw,
+	.port_set_masked_raw = gpio_xec_port_set_masked_raw,
+	.port_set_bits_raw = gpio_xec_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_xec_port_clear_bits_raw,
+	.port_toggle_bits = gpio_xec_port_toggle_bits,
+	.pin_interrupt_configure = gpio_xec_pin_interrupt_configure,
+	.manage_callback = gpio_xec_manage_callback,
+};
+
+#define XEC_GPIO_PORT_FLAGS(n)						\
+	(DT_IRQ_HAS_CELL(DT_DRV_INST(n), irq)) ? GPIO_INT_ENABLE : 0
```

Comment:
macros with complex values must be enclosed by parenthesis

Reply:
Done